src/equations/hyperbolic_diffusion_2d.jl

```diff
@@ -136,10 +138,10 @@ end
     if orientation == 1
         f1 = -equations.nu * q1
         f2 = -phi * inv_Tr
-        f3 = zero(phi)
+        f3 = 0
     else
         f1 = -equations.nu * q2
-        f2 = zero(phi)
+        f2 = 0
         f3 = -phi * inv_Tr
     end
```

Comment:
This is problematic since `f2` and `f3` will have different types depending on the branch - it's a type instability. The previous version avoids this issue.

Reply:
So true. Thanks!